Automattic/wp-calypso
# Framework: Reduxify embeds

It's been a while since we started migrating all the legacy `EventEmitter` and `flux` stores to use `redux`. Most of them have been migrated, but there are still some remaining, mostly big and thorny ones. 

This PR creates the embeds Redux infrastructure, and ports the existing embed flux stores usage to Redux. Finally, it removes those legacy flux stores. 

This removes the last usages of `flux/utils` and of flux's `ReduceStore`. A thorough list of changes can be found in the list below.

I'm happy to split it into multiple PRs, but I didn't because:
* The Redux infrastructure is pretty straightforward and simple
* A big portion of the added code is tests.
* Component reduxification goes well together rather than being split into multiple PRs.
* Commits are atomic and are supposed to provide background to the changes.

Note that I didn't create a query component for the list of embed patterns, simply because it wasn't needed.

#### Changes proposed in this Pull Request

* Introduce Redux infrastructure for handling embeds:
  * Action creators
  * Data layer
  * Reducer
  * Utils (used for normalization of embed regex patterns)
  * Selectors
  * Tests for utils, reducer and selectors - not necessary for data layer and action creators as they would be simple dupes of the code that they test.
  * Query component for the embed rendering (`QueryEmbed`)
* Reduxify the `EmbedView` tinymce plugin view component
* Reduxify the `EmbedViewManager` component
* Remove legacy flux embed stores.

#### Testing instructions

* Checkout this branch.
* Go to the post edit screen for a WP.com site.
* Make sure you're using the Calypso post editor instead of Gutenberg.
* Try inserting an embeddable URL in your post.
* Verify it gets converted to a proper embed in visual mode.
* Compare the embeds behavior with what's in production, it should work the same way.
* Make sure to test with a clean Redux store.
* Verify all tests pass.
* If you'd like to test the Redux infrastructure:
  * Open your browser console.
  * Type `dispatch( { type: 'EMBEDS_REQUEST', siteId: 1234 } )` where `1234` is the ID of one of your sites.
  * Verify there's a network request to fetch the supported embed regex patterns.
  * Type `state.embeds.siteItems` in your browser console.
  * Verify you can see the embed patterns keyed by site ID in the state.
  * Type `dispatch( { type: 'EMBED_REQUEST', siteId: 1234, url: 'https://www.facebook.com/20531316728/posts/10154009990506729/' } )` where `1234` is the ID of one of your sites.
  * Verify there's a network request to fetch the embed render data.
  * Type `state.embeds.urlItems` in your browser console.
  * Verify you can see the embed render data in the state.

Fixes #7924
